src/server/field.rs

```diff
@@ -556,10 +522,7 @@ pub trait ReadEntry: PrivReadEntry + Sized {
             return End(self);
         }
 
-        let field_headers = match try_read_entry!(self; self.read_headers()) {
-            Some(headers) => headers,
-            None => return End(self),
-        };
+        let field_headers = try_read_entry!(self; self.read_headers().map_err(|e| io::Error::new(io::ErrorKind::Other, e)));
```

Comment:
I'm not sure of the convention here, but I think `io::ErrorKind::InvalidData` is more appropriate.